### ISSUE TITLE
Loosing focus when starting python shell with elpy-shell-send-functions

### ISSUE THREAD

**Issue by bbo2adwuff** [jorgenschaefer/elpy#1848]
# Summary

With `elpy-shell-send-group-and-step` the cursor and focus should move right to the next group in the same buffer, right?
However, when the python shell still needs to be started (i.e. when `run-python` is called by elpy) then the Python shell buffer is selected.

That behaviour I would expect with `elpy-shell-send-group-and-step-and-go`.

Plus the cursor "gets stuck" right after the first `>>>` and doesn't scroll down if I don't `end-of-buffer` before switching back to the code buffer.

Same holds for any other `...-step` function (without `...-and-go`).

# Steps to reproduce

1. Open python code.
2. Run `elpy-shell-send-group-and-step`
3. Run `end-of-buffer`
4. Run `other-window`
5. Run `elpy-shell-send-group-and-step`
6. Run `elpy-shell-send-group-and-step`

Step 3-5 shouldn't be necessary.

From step 6 onwards `elpy-shell-send-group-and-step` works as expected with the focus staying in the same buffer.

(In step 5 the first group will be send again as in step 2 the cursor was not moved to the next group.)
(Seems that step 3 can be solved by `(setq comint-move-point-for-output t)`. But when the cursor is at the end of the buffer after step 3 then the output moves as expected anyways. So using `comint-move-point-for-output` seems just like a workaround for the initial state.)

# My configuration
## OS

GNU Emacs 27.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.22, cairo version 1.17.3) of 2020-08-28 on Archlinux 5.9.3-arch1-1

## Result of `(elpy-config)`
```
Emacs.............: 27.1
Elpy..............: 1.34.0
Virtualenv........: rose (/home/bbo/.virtualenvs/rose/)
Interactive Python: python 3.8.6 (/home/bbo/.virtualenvs/rose/bin/python)
RPC virtualenv....: rpc-venv (/home/bbo/.emacs.d/elpy/rpc-venv)
 Python...........: python 3.8.6 (/home/bbo/.emacs.d/elpy/rpc-venv/bin/python)
 Jedi.............: 0.17.2
 Rope.............: 0.18.0
 Autopep8.........: 1.5.4
 Yapf.............: 0.30.0
 Black............: 20.8b1
Syntax checker....: flake8 (/usr/bin/flake8)
```

## Elpy configuration in my init.el
Issue also persists when I start with this minimal init.
```
(package-initialize)
(elpy-enable)
```

**Comment by galaunay** [comment]
Yes, you are right this is not the expected behaviour. 
Thanks for the feedback.

I made a commit to fix this issue. It will be fixed in the next release (Elpy 1.35) that I intend to make in the following days.

Please re-open if there is still something wrong at this point.

**Comment by bbo2adwuff** [comment]
Perfect, the fix works like a charm! Thanks a lot!
